core/system/src/mpu/vmpu.c

```diff
@@ -59,16 +59,24 @@ static int vmpu_sanity_checks(void)
     assert(vmpu_bits(0x8001UL) == 16);
     assert(vmpu_bits(1) == 1);
 
+    /* Verify that the known hard-coded symbols are equal to the ones taken from
+     * the host linker script. */
+    assert((uint32_t) __uvisor_config.flash_start == FLASH_ORIGIN);
+    assert((uint32_t) __uvisor_config.flash_offset == FLASH_ORIGIN + RESERVED_FLASH);
+    assert((uint32_t) __uvisor_config.sram_start == SRAM_ORIGIN);
+    assert((uint32_t) __uvisor_config.sram_offset == SRAM_ORIGIN + RESERVED_SRAM);
```

Comment:
I think I didn't explain those symbols correctly. `__uvisor_config.flash_start` is supposed to be exactly `FLASH_ORIGIN` and so on, because they represent exactly the same thing.

Here I'm basically performing sanity checks between what I hardcoded in uVisor (`FLASH_ORIGIN`) and what I instead receive from the host linker script (`__uvisor_config.flash_origin`) in the configuration table. This table is populated by mbed so we need to check that it was done properly. 

If the misunderstanding was caused by `RESERVED_{FLASH, SRAM}`, they actually mean "space to reserve before uVisor is placed", so in a sense they should be called `{FLASH, SRAM}_OFFSET` just like their counterparts in the configuration table. (Spoiler alert: they will be renamed, in a subsequent PR)